server/src/main/java/org/apache/cloudstack/network/lb/LoadBalancerConfigManagerImpl.java

```diff
@@ -0,0 +1,396 @@
+import org.apache.cloudstack.context.CallContext;
+import org.apache.cloudstack.framework.config.ConfigKey;
+import org.apache.cloudstack.network.lb.LoadBalancerConfig.Scope;
+import org.apache.log4j.Logger;
+
+public class LoadBalancerConfigManagerImpl extends ManagerBase implements LoadBalancerConfigService, LoadBalancerConfigManager {
+    private static final Logger LOGGER = Logger.getLogger(LoadBalancerConfigManagerImpl.class);
+
+    @Inject
+    LoadBalancerConfigDao _lbConfigDao;
+    @Inject
+    NetworkDao _networkDao;
+    @Inject
+    VpcDao _vpcDao;
+    @Inject
+    LoadBalancerDao _lbDao;
+    @Inject
+    AccountManager _accountMgr;
+    @Inject
+    LoadBalancingRulesManager _lbMgr;
+    @Inject
+    NetworkServiceMapDao _ntwkSrvcDao;
+
+    @Override
+    public List<? extends LoadBalancerConfig> searchForLoadBalancerConfigs(ListLoadBalancerConfigsCmd cmd) {
+        Long id = cmd.getId();
+        String scopeStr = cmd.getScope();
+        Long networkId = cmd.getNetworkId();
+        Long vpcId = cmd.getVpcId();
+        Long loadBalancerId = cmd.getLoadBalancerId();
+        String name = cmd.getName();
+
+        if (id == null && scopeStr == null) {
+            throw new InvalidParameterValueException("At least one of id/scope is required");
+        }
+
+        //validate parameters
+        Scope scope = null;
+        if (scopeStr != null) {
+            scope = LoadBalancerConfigKey.getScopeFromString(scopeStr);
+            if (scope == null) {
+                throw new InvalidParameterValueException("Invalid scope " + scopeStr);
+            }
+            checkPermission(scope, networkId, vpcId, loadBalancerId, cmd.listAll());
+        }
+
+        if (id != null) {
+            LoadBalancerConfigVO config = _lbConfigDao.findById(id);
+            if (config == null) {
+                throw new InvalidParameterValueException("Cannot find load balancer config by id " + id);
+            }
+            checkPermission(config);
+        }
+
+        if (cmd.listAll()) {
+            if (id != null || name != null) {
+                throw new InvalidParameterValueException("id and name must be null if listall is true");
+            }
+        }
+
+        SearchCriteria<LoadBalancerConfigVO> sc = _lbConfigDao.createSearchCriteria();
+        if (id != null) {
+            sc.addAnd("id", SearchCriteria.Op.EQ, id);
+        }
+        if (scope != null) {
+            sc.addAnd("scope", SearchCriteria.Op.EQ, scope);
+        }
+        if (networkId != null) {
+            sc.addAnd("networkId", SearchCriteria.Op.EQ, networkId);
+        }
+        if (vpcId != null) {
+            sc.addAnd("vpcId", SearchCriteria.Op.EQ, vpcId);
+        }
+        if (loadBalancerId != null) {
+            sc.addAnd("loadBalancerId", SearchCriteria.Op.EQ, loadBalancerId);
+        }
+        if (name != null) {
+            sc.addAnd("name", SearchCriteria.Op.EQ, name);
+        }
```

Comment:
This code seems repeated, but changing parameters; We could extract it to a method or create an `addAndNotNull` to `SearchCriteria`.